pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -95,12 +101,12 @@ export namespace yaml {
         {{#Groups}}
         {{#Versions}}
         {{#Kinds}}
-        public getResource(groupVersionKind: "{{RawAPIVersion}}/{{Kind}}", name: string): {{Group}}.{{Version}}.{{Kind}};
-        public getResource(groupVersionKind: "{{RawAPIVersion}}/{{Kind}}", namespace: string, name: string): {{Group}}.{{Version}}.{{Kind}};
+        public getResource(groupVersionKind: "{{RawAPIVersion}}/{{Kind}}", name: string): pulumi.Output<{{Group}}.{{Version}}.{{Kind}}>;
+        public getResource(groupVersionKind: "{{RawAPIVersion}}/{{Kind}}", namespace: string, name: string): pulumi.Output<{{Group}}.{{Version}}.{{Kind}}>;
         {{/Kinds}}
         {{/Versions}}
         {{/Groups}}
-        public getResource(groupVersionKind: string, namespaceOrName: string, name?: string): pulumi.CustomResource {
+        public getResource(groupVersionKind: string, namespaceOrName: string, name?: string): pulumi.Output<pulumi.CustomResource> {
```

Comment:
Note that this is a breaking change - and will also be a bit more cumbersome to use in practice.  It's also a bit of a shame, because `CustomResource` objects are already delayed.  Not sure there is anything we can do here in the nearterm to handle this better - but another case to think about as part of https://github.com/pulumi/pulumi/issues/994.